olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -288,6 +320,77 @@ model::Partition PartitionsRepository::PartitionFromSubQuad(
   return ret;
 }
 
+PartitionsResponse PartitionsRepository::GetAggregatedPartitionForVersionedTile(
+    client::HRN catalog, std::string layer, client::CancellationContext context,
+    const TileRequest& request, int64_t version,
+    const client::OlpClientSettings& settings) {
+  const auto fetch_option = request.GetFetchOption();
+  const auto& tile_key = request.GetTileKey();
+  const auto& root_tile_key = tile_key.ChangedLevelBy(-kAggregateQuadTreeDepth);
```

Comment:
This is not correct. If you are doing it like this we will have endless duplicates in the cache. E.g. user requests tile at level 14, you download quadtree at level 10, which includes levels {14, 13, 12, 11, 10} and parents up to level 1.
Now the user requests a level 13 tile. So you go on level 9 and request the quadtree index, which includes levels {13, 12, 11, 10, 9}. But you already have these partitions in the level 10 quadtree index. Why do you download it again?

Reply:
But strategy was to use quad trees with depth 4, so they are different for tiles from different levels.
Quad tree will be the same for 256 level 14 tiles.
I suppose this is actual use case for GetAggregatedData() - request neighbor tiles from same level. Am I wrong?

Reply:
Yes, and it still is to request with depth 4, but not if we already have it in cache within one of the quadtrees that were downloaded before.